va_explorer/utils/file_io.py

```diff
@@ -0,0 +1,52 @@
+import csv
+from django.http import HttpResponse
+
+
+def download_queryset_as_csv(queryset, filename, location):
+    # Set the response details
+    response = csv_response_with_redirect(filename, location)
+
+    model = queryset.model
+    model_fields = model._meta.fields + model._meta.many_to_many
+    field_names = [field.name for field in model_fields]
+
+    writer = csv.writer(response, delimiter=",")
+    # Write the header information
+    writer.writerow(field_names)
+    # Write the data rows
+    for row in queryset:
+        values = []
+        for field in field_names:
+            value = getattr(row, field)
+            if callable(value):
+                try:
+                    value = value() or ''
+                except:
+                    value = 'Error retrieving value'
+            if value is None:
+                value = ''
+            values.append(value)
+        writer.writerow(values)
+
+    return response
+
+def download_list_as_csv(generic_list, filename, location):
+    # Set the response details
+    response = csv_response_with_redirect(filename, location)
+
+    writer = csv.writer(response, delimiter=",")
+    for item in generic_list:
+        # Note .writerow() requires a sequence ('', (), []) and places each index in its own column of the row, sequentially
+        # If your desired string is not an item in a sequence, writerow() will iterate over each letter in your string
+        writer.writerow([item])
+
+    return response
+
+def csv_response_with_redirect(filename, location):
+    response = HttpResponse(content_type='text/csv')
+    response['Location'] = location
+    # TODO: This does not currently set the filename
```

Comment:
@ajbarnes I think I came across that stackoverflow post as well at some point when developing. 

Yes, I'm wondering how you feel about opening another small issue to address the filename, rather than doing it in this PR.

Reply:
Yeah, as part of the cleanup towards the end of March, I imagine creating issues for all the leftover TODOs will be in order.